### PR TITLE
Up to 0.18.0 (added feature "serde-support")

### PR DESCRIPTION
**Pull request by DmitryAstafyev**
- Upgrade versions of some outdated dev-dependencies
- Exclude `Serialize` and `Deserialize` into feature `serde-support`
- Resolve clippy warnings

### CODE REVIEW

**Review by kruss — APPROVED** [review]
Some minor comments which could be considered without another review.

File: src/dlt.rs [hunk 1]
```diff
@@ -59,7 +66,11 @@ pub struct Message {
 }
 
 /// Add some stupid docs
```
[the remark this answers is not in the document]
Comment (DmitryAstafyev) [reply src/dlt.rs:68]: O_o

File: src/filtering.rs [hunk 2]
```diff
@@ -91,10 +94,10 @@ impl From<&DltFilterConfig> for ProcessedDltFilterConfig {
...
 }
 
-/// Read filter config from a json file
-pub fn read_filter_options(f: &mut fs::File) -> Option<DltFilterConfig> {
-    let mut contents = String::new();
-    f.read_to_string(&mut contents)
-        .ok()
-        .and_then(|_| serde_json::from_str(&contents[..]).ok())
-}
+// /// Read filter config from a json file
```
[the remark this answers is not in the document]
Comment (DmitryAstafyev) [reply src/filtering.rs]: Many thanks for your review... I've commented it and forget to uncomment O_o

**Review by marcmo — APPROVED** [review]
LGTM

**Pull request merged**